src/client/datascience/interactive-ipynb/nativeEditorCommandListener.ts

```diff
@@ -100,14 +99,18 @@ export class NativeEditorCommandListener implements IDataScienceCommandListener
         }
     }
 
-    private interruptKernel(document: NotebookDocument | undefined) {
+    private interruptKernel(document: Uri | undefined) {
         // `document` may be undefined if this command is invoked from the command palette.
-        const target =
-            this.provider.activeEditor?.file.toString() === document?.uri.toString()
-                ? this.provider.activeEditor
-                : this.provider.editors.find((editor) => editor.file.toString() === document?.uri.toString());
-        if (target) {
-            target.interruptKernel().ignoreErrors();
+        if (document) {
+            const target =
+                this.provider.activeEditor?.file.toString() === document.toString()
+                    ? this.provider.activeEditor
+                    : this.provider.editors.find((editor) => editor.file.toString() === document.toString());
+            if (target) {
+                target.interruptKernel().ignoreErrors();
+            }
+        } else {
+            this.provider.activeEditor?.interruptKernel().ignoreErrors();
```

Comment:
Do we need this same change for restart kernel?

Reply:
I don't think so, that one goes directly to the activeEditor.